### DISCUSSION TITLE
guidance for implementation of a house structure.

### DISCUSSION THREAD

**Discussion by manuel2001bk** [pmndrs/react-three-csg#32] [General]
I have a problem since I need to implement the structure of a house to then perform the subtraction of the windows and doors, only that the structure is quite complex when using group within groups and then the mesh to generate the walls and so on.

The question is what would be the ideal way to generate the model and then add the subtraction of the windows and doors.

I currently design this structure but it does not work for me as it should be as it is originally generated

The structure of the code to generate the original house is the following:

`<group ref={group} dispose={null} {...props}>
        <group position={scene.position}
            scale={scene.scale}
            rotation={scene.rotation}>
            <group position={walls.position}
                scale={walls.scale}
                rotation={walls.rotation}>
                {walls.children.map((child, index) => {
                    return <Select enabled={selected === index ? true : false}>
                        <mesh
                            key={index}
                            onClick={() => { selectedWall(index); setSelected(index) }}
                            position={child.position}
                            scale={child.scale}
                            rotation={child.rotation}
                            geometry={child.geometry}
                            material={child.material} />
                    </Select>
                })}
            </group>
            <group position={roofs.position}
                scale={roofs.scale}
                rotation={roofs.rotation}>
                {roofs.children.map((child, index) => {
                    return <mesh
                        key={index}
                        position={child.position}
                        scale={child.scale}
                        rotation={child.rotation}
                        geometry={child.geometry}
                        material={child.material} />
                })}
            </group>
        </group>
    </group>`

**Comment by MaartenBreeedveld** [comment]
The problem is not the complexity here, you can easiliy have 5 times the amount of components and not bump into any performance issues. The problem is that you just want to try and put everything into one component.

You should start by breaking it up into logical groups thinking about which component should be responisble for what task.

In Applications like Revit, It's structured by level. So you might want to do the same :)
